cypress/e2e/sliding-sync/sliding-sync.ts

```diff
@@ -0,0 +1,333 @@
+        });
+    };
+    const bumpRoom = (alias: string) => {
+        // Send a message into the given room, this should bump the room to the top
+        cy.get<string>(alias).then((roomId) => {
+            return cy.sendEvent(roomId, null, "m.room.message", {
+                body: "Hello world",
+                msgtype: "m.text",
+            });
+        });
+    };
+    const createAndJoinBob = () => {
+        // create a Bob user
+        let baseUrl;
+        let creds;
+        cy.get<SynapseInstance>("@synapse").then((synapse) => {
+            baseUrl = synapse.baseUrl;
+            return cy.registerUser(synapse, "bob", "passwordgoeshere123", "Bob");
+        }).then((c) => {
+            creds = c;
+            return cy.window({ log: false });
+        }).then((window) => {
+            return window.matrixcs.createClient({
+                baseUrl: baseUrl,
+                accessToken: creds.accessToken,
+                userId: creds.userId,
+                deviceId: creds.deviceId,
+            });
+        }).as("bob");
+
+        // invite Bob to Test Room and accept then send a message.
+        cy.all([cy.get<string>("@roomId"), cy.get<MatrixClient>("@bob")]).then(([roomId, bob]) => {
+            return cy.inviteUser(roomId, bob.getUserId()).then(() => {
+                return bob.joinRoom(roomId);
+            });
+        });
+    };
+
+    // sanity check everything works
+    it("should correctly render expected messages", () => {
+        cy.get<string>("@roomId").then(roomId => cy.visit("/#/room/" + roomId));
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.IRC);
+
+        // Wait until configuration is finished
+        cy.contains(
+            ".mx_RoomView_body .mx_GenericEventListSummary .mx_GenericEventListSummary_summary",
+            "created and configured the room.",
+        );
+
+        // Click "expand" link button
+        cy.get(".mx_GenericEventListSummary_toggle[aria-expanded=false]").click();
+    });
+
+    it("should render the Rooms list in reverse chronological order by default and allowing sorting A-Z", () => {
+        // create rooms and check room names are correct
+        cy.createRoom({ name: "Apple" }).then(() => cy.contains(".mx_RoomSublist", "Apple"));
+        cy.createRoom({ name: "Pineapple" }).then(() => cy.contains(".mx_RoomSublist", "Pineapple"));
+        cy.createRoom({ name: "Orange" }).then(() => cy.contains(".mx_RoomSublist", "Orange"));
+        // check the rooms are in the right order
+        cy.get(".mx_RoomTile").should('have.length', 4); // due to the Test Room in beforeEach
+        checkOrder([
+            "Orange", "Pineapple", "Apple", "Test Room",
+        ]);
+
+        cy.contains(".mx_RoomSublist", "Rooms").find(".mx_RoomSublist_menuButton").click({ force: true });
+        cy.contains("A-Z").click();
+        cy.get('.mx_StyledRadioButton_checked').should("contain.text", "A-Z");
```

Comment:
I would recommend to use `contains`:

```suggestion
        cy.contains(".mx_StyledRadioButton_checked", "A-Z");
```

`cy.get…` = find `.mx_StyledRadioButton_checked` then check if it contains `A-Z`  
`cy.contains…` =  find an `.mx_StyledRadioButton_checked` that contains `A-Z`

Reply:
They both boil down to the same thing due to https://docs.cypress.io/guides/core-concepts/retry-ability

Reply:
Thanks for the info. I've read through it again.
Updated the comment to avoid confusion :laughing: 

Anyway I found their tip :wink: 

![image](https://user-images.githubusercontent.com/6216686/186850043-9fec3ebd-95fd-4454-bb76-f2c2bbb0e975.png)